From: Steve Peters <scpeters@openrobotics.org>
Date: Tue, 27 Oct 2020 00:28:40 -0700
Subject: SimpleTrackedVehiclePlugin: fix for boost 1.74 (#2862)

Disable some code in SimpleTrackedVehiclePlugin for boost 1.74
or greater due to changes in boost shared_ptr.hpp

Signed-off-by: Steve Peters <scpeters@openrobotics.org>
---
 plugins/SimpleTrackedVehiclePlugin.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/SimpleTrackedVehiclePlugin.cc b/plugins/SimpleTrackedVehiclePlugin.cc
index 3be4776..c5a22b0 100644
--- a/plugins/SimpleTrackedVehiclePlugin.cc
+++ b/plugins/SimpleTrackedVehiclePlugin.cc
@@ -18,6 +18,8 @@
 #include <functional>
 #include <vector>
 
+#include <boost/version.hpp>
+
 #include <ignition/common/Profiler.hh>
 #include <ignition/math/Vector3.hh>
 #include <ignition/math/Pose3.hh>
@@ -30,6 +32,7 @@
 
 #include "plugins/SimpleTrackedVehiclePlugin.hh"
 
+#if BOOST_VERSION < 107400
 namespace std {
 template<class T>
 class hash<boost::shared_ptr<T>> {
@@ -38,6 +41,7 @@ class hash<boost::shared_ptr<T>> {
   }
 };
 }
+#endif
 
 namespace gazebo
 {
